pennylane/transforms/adjoint_metric_tensor.py

```diff
@@ -20,41 +20,20 @@
 import pennylane as qml
 from pennylane import numpy as np
 
-# pylint: disable=protected-access
+# pylint: disable=too-many-statements
 from pennylane.transforms.metric_tensor import _contract_metric_tensor_with_cjac
 
 
-def _apply_operations(state, op, device, invert=False):
-    """Wrapper that allows to apply a variety of operations---or groups
-    of operations---to a state or to prepare a new state.
-    If ``invert=True``, this function makes sure not to alter the operations.
-    The state of the device, however may be altered, depending on the
-    device and performed operation(s).
-    """
-    # pylint: disable=protected-access
-    if isinstance(op, (list, np.ndarray)):
-        if invert:
-            op = op[::-1]
-        for _op in op:
-            state = _apply_operations(state, _op, device, invert)
-        return state
-
-    if isinstance(op, qml.QubitStateVector):
-        if invert:
-            raise ValueError("Can't invert state preparation.")
-        device._apply_state_vector(op.parameters[0], op.wires)
-        return device._state
-
-    if isinstance(op, qml.BasisState):
-        if invert:
-            raise ValueError("Can't invert state preparation.")
-        device._apply_basis_state(op.parameters[0], op.wires)
-        return device._state
+def _get_op_for_generator(generator, like_real):
+    mat = qml.matrix(generator)
+    real_mat = qml.math.convert_like(mat, like_real)
+    return qml.QubitUnitary(real_mat, generator.wires)
 
-    apply_op = qml.adjoint(op) if invert else op
-    state = device._apply_operation(state, apply_op)
 
-    return state
+def _reshape_real_imag(state, dim):
+    real_part = qml.math.reshape(qml.math.real(state), (dim,))
+    imag_part = qml.math.reshape(qml.math.imag(state), (dim,))
+    return real_part, imag_part
```

Comment:
Probably not important, but could also do with one reshape? :thinking: 
```suggestion
    state = qml.math.reshape(state, (dim,))
    return qml.math.real(state), qml.math.imag(state)
```

Reply:
Probably. Was just copy and pasting what already there.